platform/engineering/design-docs/2020-05-26-content-validation.md

```diff
@@ -0,0 +1,388 @@
+
+##### Accessibility errors
+
+###### Positive
+
+1. Confirm that the content build has zero accessibility errors
+2. Run the content validation script
+3. Confirm that the script reports zero accessibility errors
+
+###### Negative
+
+1. Confirm that the content build has zero accessibility errors
+2. Modify an HTML file, like [`header.html`](https://github.com/department-of-veterans-affairs/content-build/blob/master/src/site/includes/header.html), to have an accessibility error
+3. Confirm that the script reports the newly introduced accessibility error
+
+#### CI
+
+##### Positive
+
+1. Confirm that the content build has zero accessibility errors and zero broken links
+2. Run the job that calls the content validation script
+3. Confirm that a report is sent to the #cms-team Slack channel
+4. Confirm that the report includes zero accessibility errors and zero broken links
+
+##### Negative
+
+1. Confirm that the content build has zero accessibility errors and zero broken links
+2. Add a broken link, and an accessibility error
+3. Run the job that calls the content validation script
+4. Confirm that a report is sent to the #cms-team Slack channel
+5. Confirm that the report includes the newly introduced accessibility error
+6. Confirm that the report includes the newly introduced broken link
+7. Fix the newly introduced accessibility error
+8. Fix the newly introduced broken link
+9. Run the job that calls the content validation script
+10. Confirm that the report includes zero accessibility errors and zero broken links
+
+### Logging
+
+Currently, invalid content is logged in the build log during staging deploys. And failed builds trigger generic [Slack notifications](https://dsva.slack.com/archives/C37M86Y8G/p1591099447374800).
+
+The updated content validation approach will continue logging invalid content errors in the build log, and will also post an automated report of the invalid content in the #cms-team Slack channel.
+
+### Debugging
+
+- Local build logs
+- Jenkins build logs
+- Slack notifications
+
+### Caveats
+
+- The current broken link checking in Drupal can handle redirects, but those redirects are not currently synchronized with `vets-website`. When the redirects are not synchronized between `vets-website` and Drupal, a page that is moved in Drupal may pass the Drupal broken link check but fail the `vets-website` broken link check.
+- Many of the broken links come from URLs that change, according to decisions by the Content & IA team, usually in conjunction with Public Websites. In that workflow, an engineer makes a code change for the redirect, and has to time the release of the redirect code to match the content release of the URL change in the content layer. It's a fragile and very time-consuming process, and will likely not be scalable as we add hundreds of editors from outside the tight Github/Slack communication flows (e.g. Public Affairs Officers at 2000+ VA facilities).
```

Comment:
(nit) Also doesn't really feel like a caveat, maybe background info?

Reply:
Moved from caveat to background in https://github.com/department-of-veterans-affairs/va.gov-team/pull/9658/commits/152e818308af23b7148b4acbec1de630e52fe6fd